packages/amplify-category-api/src/provider-utils/awscloudformation/utils/graphql-schema-utils.ts

```diff
@@ -31,8 +34,16 @@ export const generateRDSSchema = async (
   try {
     await adapter.initialize();
   } catch(error) {
-    printer.error('Failed to connect to the specified RDS Data Source. Check the connection details in the schema and re-try. Use "amplify api update-secrets" to update the user credentials.');
-    throw(error);
+    // If connection is unsuccessful, try connecting from VPC
+    if (error.code === 'ETIMEDOUT') {
+      const canConnectFromVpc = await retryWithVpcLambda(context, databaseConfig, adapter);
+      if (!canConnectFromVpc) {
+        throw new Error(UNABLE_TO_CONNECT_MESSAGE);
```

Comment:
This should be an `AmplifyError` I think, or is this being caught upstream by the transformer error interceptor?